readthedocs/projects/forms.py

```diff
@@ -61,10 +61,12 @@ def __init__(self, *args, **kwargs):
         )
         # If there is a remote repo attached to the project, add it to the queryset,
         # since the current user might not have access to it.
-        if current_remote_repo:
-            queryset |= RemoteRepository.objects.filter(
-                pk=current_remote_repo.pk
-            ).distinct()
+        if current_remote_repo and current_remote_repo not in queryset:
+            # NOTE: we are not including the current remote repo in the queryset
+            # using an "or" condition, that confuses the ORM/postgres and
+            # it results in a very slow query. We are going to evaluate the
+            # queryset anyway.
+            queryset = [current_remote_repo] + list(queryset)
```

Comment:
this isn't as simple as passing a list instead of a queryset, it's calling `.all()` internally :/, but I think I found another easy way